eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/tcp/impl/eventmeshmessage/EventMeshMessageTCPSubClient.java

```diff
@@ -73,9 +73,11 @@ public void reconnect() throws EventMeshException {
             super.reconnect();
             hello();
             if (!CollectionUtils.isEmpty(subscriptionItems)) {
-                for (SubscriptionItem item : subscriptionItems) {
-                    Package request = MessageUtils.subscribe(item.getTopic(), item.getMode(), item.getType());
-                    this.io(request, EventMeshCommon.DEFAULT_TIME_OUT_MILLS);
+                synchronized (subscriptionItems) {
+                    for (SubscriptionItem item : subscriptionItems) {
```

Comment:
I think it should use synchinized key word in subscribe() to protect subscriptionItems .

Reply:
What's you mean that add synchronize key word in subscribe() method in current class or MessageUtils.subscribe() ?

Reply:
yes